rasa_sdk/knowledge_base/actions.py

```diff
@@ -0,0 +1,372 @@
+# -*- coding: utf-8 -*-
+from rasa_sdk import Action
+from rasa_sdk.events import SlotSet
+from rasa_sdk.knowledge_base.storage import (
+    SCHEMA_KEYS_ATTRIBUTES,
+    SCHEMA_KEYS_REPRESENTATION,
+    SCHEMA_KEYS_KEY,
+)
+
+
+SLOT_MENTION = "mention"
+SLOT_OBJECT_TYPE = "object_type"
+SLOT_ATTRIBUTE = "attribute"
+SLOT_LISTED_OBJECTS = "knowledge_base_listed_objects"
+SLOT_LAST_OBJECT = "knowledge_base_last_object"
+SLOT_LAST_OBJECT_TYPE = "knowledge_base_last_object_type"
+
+
+class ActionKnowledgeBase(Action):
```

Comment:
it seems to me that it is not really an `Action` but rather a collection of helper methods, because it doesn't have `run` method. I'd rather merge this with another action. Even though it'll be very large, I think it'll be less confusing for a user

Reply:
I kept it separate so that users could inherit this Action and implement more use cases. That way they could use our default implementation along their one implementations.

Reply:
but cannot they inherit the other one and implement their implementation?

Reply:
yeah... but I actually don't like that. I guess it is more clear if you keep it separate.

Reply:
I see, ok. Can we add some run method there? I think it's a bit not clear how to use it on its own

Reply:
I removed it as @wochinge pointed out that they are not needed. But can add them again.

Reply:
I don't know, but if this action can be used by its own, I'd put `run` action there

Reply:
Removed the abstract action and added all helper functions to a `utils` package.